src/openrct2/object/ObjectManager.cpp

```diff
@@ -503,10 +503,6 @@ class ObjectManager final : public IObjectManager
                 sgObject->UpdateEntryIndexes();
             }
         }
-
-        // HACK Scenery window will lose its tabs after changing the scenery group indexing
-        //      for now just close it, but it will be better to later tell it to invalidate the tabs
-        WindowCloseByClass(WindowClass::Scenery);
```

Comment:
I _think_ this hack is no longer necessary, owing to #23621. We could choose to keep it around, but then I'd need to find a way to prevent it from being called while UiContext hasn't been initialised yet. Otherwise, ironically, it leads to a crash.

Reply:
I suggest we/you just test it by changing the selection and see if it mucks up the window.

Reply:
The scenery window closes as soon as I open the object selection window. If I open the former again while the latter is open, it stays open. As soon as I select a scenery group, though, the scenery window is closed immediately. I'm not sure by what mechanism it is closed, but the removal of this hack does not seem to have affecting it.